### PR TITLE
:sparkles: Support passing identifying parameters for zaaktype & resu…

### PR DESCRIPTION
**Pull request by damm89**
…ltaattype

### CODE REVIEW

File: src/bptl/work_units/zgw/tasks/resultaat.py [hunk 1]
```diff
@@ -18,10 +20,12 @@ class CreateResultaatTask(ZGWWorkUnit):
     **Required process variables**
 
     * ``zaakUrl``: full URL of the ZAAK to set the RESULTAAT for
-    * ``resultaattype``: full URL of the RESULTAATTYPE to set
     * ``bptlAppId``: the application ID of the app that caused this task to be executed.
       The app-specific credentials will be used for the API calls.
     * ``services``: DEPRECATED - support will be removed in 1.1
+    * ``resultaattype``: full URL of the RESULTAATTYPE to set
+    -OR-
+    * ``omschrijving``: description of RESULTAAT.
```
Comment (sergei-maertens) [inline src/bptl/work_units/zgw/tasks/resultaat.py:28]: *Please* check the documentation output on http://localhost:8000/taskmappings/add/, this is causing Sphinx errors now:

```
System Message: WARNING/2 (<string>, line 13)

Bullet list ends without a blank line; unexpected unindent.
```

File: src/bptl/work_units/zgw/tasks/resultaat.py [hunk 2]
```diff
@@ -36,22 +40,61 @@ class CreateResultaatTask(ZGWWorkUnit):
...
+                if len(resultaattype) != 1:
+                    raise ValueError(
+                        "No%s resultaattype was found with zaaktype %s and omschrijving %s."
+                        % (
+                            "" if len(resultaattype) == 0 else " unique",
+                            zaak["zaaktype"],
+                            omschrijving,
+                        )
+                    )
+                resultaattype = resultaattype[0]["url"]
```
Comment (sergei-maertens) [inline src/bptl/work_units/zgw/tasks/resultaat.py:76]: This belongs in the `else` part of

```
try:
     ...
except ...:
    ...
else:
    ...
```

The entire code block after the `check_variable` should not raise a `MissingVariable`, so it should be outside of this specific try-except block.

File: src/bptl/work_units/zgw/tasks/resultaat.py [hunk 3]
```diff
@@ -36,22 +40,61 @@ class CreateResultaatTask(ZGWWorkUnit):
...
+                    raise ValueError(
+                        "No resultaattypen were found for zaaktype %s."
+                        % zaak["zaaktype"]
+                    )
+                resultaattype = list(
+                    filter(
+                        lambda rt: rt["omschrijving"].lower() == omschrijving.lower(),
+                        resultaattypen["results"],
+                    )
+                )
```
Comment (sergei-maertens) [inline src/bptl/work_units/zgw/tasks/resultaat.py:66]: that's not pythonic, please use a list-comprehension instead - much more readable

File: src/bptl/work_units/zgw/tasks/zaak.py [hunk 4]
```diff
@@ -30,46 +32,123 @@ class CreateZaakTask(ZGWWorkUnit):
...
+                            zaaktype.begin_geldigheid
+                            <= date.today()
+                            <= zaaktype.einde_geldigheid
+                        )
+                    else:
+                        return zaaktype.begin_geldigheid <= date.today()
+
+                zaaktypen = list(
+                    filter(lambda zt: _filter_on_geldigheid(zt), zaaktypen)
+                )
```
Comment (sergei-maertens) [inline src/bptl/work_units/zgw/tasks/zaak.py:126]:
```suggestion
                zaaktypen = [zt for zt in zaaktypen if _filter_on_geldigheid(zt)]
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/bptl/work_units/zgw/tasks/resultaat.py (hunk 2 above)
Comment (damm89) [reply src/bptl/work_units/zgw/tasks/resultaat.py:76]: Thanks!

File: src/bptl/work_units/zgw/tasks/resultaat.py (hunk 1 above)
Comment (damm89) [reply src/bptl/work_units/zgw/tasks/resultaat.py:28]: Good catch. Had checked it, but made some edits and apparently didn't check again.

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**